src/drivers/intel/cavs/interrupt.c

```diff
@@ -213,13 +213,13 @@ uint32_t platform_interrupt_get_enabled(void)
 
 void platform_interrupt_set(uint32_t irq)
 {
-	if (!interrupt_get_parent(irq))
+	if (interrupt_is_dsp(irq))
```

Comment:
Lets make this more meaningful, interrupt is_dsp_direct() implies it's a direct (no external IP) IRQ type.